src/GrafanaDashboardDeploy/src/DeployTool.cs

```diff
@@ -0,0 +1,152 @@
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.Services.AppAuthentication;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace DotNet.Grafana
+{
+    public static class DeployTool
+    {
+        public static async Task PostExportPack(GrafanaClient grafanaClient, string inputFilePath)
+        {
+            var folderIdMap = new Dictionary<string, int>();
+
+            using var sr = new StreamReader(inputFilePath);
+            using var jr = new JsonTextReader(sr);
+            var data = await JObject.LoadAsync(jr).ConfigureAwait(false);
+
+            foreach (var folder in data["folders"])
+            {
+                string folderUid = folder["uid"].Value<string>();
+                string folderTitle = folder["title"].Value<string>();
+
+                var result = await grafanaClient.CreateFolderAsync(folderUid, folderTitle).ConfigureAwait(false);
+                int folderId = result["id"].Value<int>();
+
+                folderIdMap.Add(folderUid, folderId);
+            }
+
+            foreach (var datasource in data["datasources"])
+            {
+                foreach (var secretData in datasource["secureJsonData"].Children<JProperty>())
+                {
+                    string secretExpression = secretData.Value.ToString();
+                    string secretName;
+                    string secretValue;
```

Comment:
You can join this with the assignment on line 48, since it's not used before hand, save some lines... actually, it's only used to set "secretData.Value", we probably can just do `secretData.Value = await GetSecretAsync(...`